### ISSUE TITLE
Make re-usable charts for use across tables/SQL queries of the same schema

### ISSUE THREAD

**Issue by asg017** [datasette/datasette-plot#2]
We should have some option where you can "save" a chart's configuration to re-use in other places. 

For example, you have have some scatterplot/linear regression visualization that you want to re-use across your Datasette instance - say one chart per month. Currently, you'll have to re-build the chart manually for every new data source (table or SQL query), but that can get tiresome. 

This will likely use Datasette's new [internal database](https://docs.datasette.io/en/latest/internals.html#datasette-s-internal-database) to persist chart components across restarts.

**Comment by tomsing1** [comment]
I _think_ that applying a filter to the dataset currently resets the plot? It would be great if the plot definition could persist across changes,  because then we can drill down into the dataset while being able to directly see the effect of filters on the (same) plot.

**Comment by asg017** [comment]
@tomsing1 yup, that's a current limitation!  Though I think we can listen for form submissions and inject query parameters on submit, which means persisting across changes. I'll give it a shot!

**Comment by ptsefton** [comment]
It would be good to link this to canned queries, so you can have a pre-configured plot associated with a query with an additional "plot" parameter in the config file.
